src/actions/walletConnectActions.js

```diff
@@ -18,595 +18,348 @@
-  connector,
-});
-
-const walletConnectCallRequest = (request: CallRequest): WalletConnectCallRequest => ({
-  type: WALLETCONNECT_CALL_REQUEST,
-  request,
-});
-
-const walletConnectSessionDisconnected = (connector: Connector): WalletConnectSessionDisconnected => ({
-  type: WALLETCONNECT_SESSION_DISCONNECTED,
-  connector,
-});
-
-const walletConnectSessionsKilled = (connectors: Connector[]): WalletConnectSessionsKilled => ({
-  type: WALLETCONNECT_SESSIONS_KILLED,
-  connectors,
-});
-
-const walletConnectCallRejected = (callId: number): WalletConnectCallRejected => ({
-  type: WALLETCONNECT_CALL_REJECTED,
-  callId,
-});
-
-const walletConnectCallApproved = (callId: number): WalletConnectCallApproved => ({
-  type: WALLETCONNECT_CALL_APPROVED,
-  callId,
-});
-
-const walletConnectSessionReceived = (): WalletConnectSessionReceived => ({
-  type: WALLETCONNECT_SESSION_RECEIVED,
-});
-
-const onWalletConnectCallRequest = (connector: Connector, payload: JsonRpcRequest) => {
-  return async (dispatch: Dispatch) => {
-    const {
-      icons, name, url,
-    } = connector.peerMeta || {
-      icons: [], name: '', url: '',
-    };
 
-    const request: CallRequest = {
-      name,
-      url,
-      icon: get(icons, '[0]'),
-      callId: payload.id,
-      peerId: connector.peerId,
-      method: payload.method,
-      params: payload.params,
-    };
 
-    dispatch(walletConnectCallRequest(request));
-
-    const navParams = {
-      callId: request.callId,
-      method: request.method,
-    };
-
-    if (!isNavigationAllowed()) {
-      updateNavigationLastScreenState({
-        lastActiveScreen: WALLETCONNECT_CALL_REQUEST_SCREEN,
-        lastActiveScreenParams: navParams,
-      });
-      return;
-    }
+const setWalletConnectErrorAction = (message: string) => {
+  return (dispatch: Dispatch) => {
+    dispatch({ type: SET_WALLETCONNECT_REQUEST_ERROR, payload: { message } });
 
-    const navigateToAppAction = NavigationActions.navigate({
-      routeName: WALLETCONNECT_CALL_REQUEST_SCREEN,
-      params: navParams,
+    Toast.show({
+      message: message || t('toast.walletConnectFailed'),
+      emoji: 'eyes',
+      supportLink: true,
+      autoClose: false,
     });
-
-    navigate(navigateToAppAction);
   };
 };
 
-const subscribeToEvents = (connector: Connector) => {
-  return async (dispatch: Dispatch) => {
-    connector.on(CALL_REQUEST_EVENT, (e: any, payload: JsonRpcRequest) => {
-      if (e) {
-        dispatch(walletConnectError(CALL_REQUEST_ERROR, e.toString()));
-
-        return;
-      }
-
-      dispatch(onWalletConnectCallRequest(connector, payload));
-    });
+export const resetWalletConnectConnectorRequestAction = (rejectedRequest: boolean = false) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const { walletConnect: { connectorRequest } } = getState();
 
-    connector.on(DISCONNECT_EVENT, (e: any) => {
-      if (e) {
-        dispatch(walletConnectError(DISCONNECT_ERROR, e.toString()));
```

Comment:
I changed this into regular action which retrieves `message` part. According to `WalletConnect` here it might return regular `Error` so `toString` is nothing but same message.